ArchiSteamFarm/IPC/Startup.cs

```diff
@@ -97,44 +100,41 @@ internal sealed class Startup {
 		// Add support for default root path redirection (GET / -> GET /index.html), must come before static files
 		app.UseDefaultFiles();
 
-		// Add support for static files (e.g. HTML, CSS and JS from IPC GUI)
-		app.UseStaticFiles(
-			new StaticFileOptions {
-				OnPrepareResponse = static context => {
-					if (context.File is { Exists: true, IsDirectory: false } && !string.IsNullOrEmpty(context.File.Name)) {
-						string extension = Path.GetExtension(context.File.Name);
-
-						CacheControlHeaderValue cacheControl = new();
-
-						switch (extension.ToUpperInvariant()) {
-							case ".CSS" or ".JS":
-								// Add support for SRI-protected static files
-								// SRI requires from us to notify the caller (especially proxy) to avoid modifying the data
-								cacheControl.NoTransform = true;
-
-								goto default;
-							default:
-								// Instruct the caller to always ask us first about every file it requests
-								// Contrary to the name, this doesn't prevent client from caching, but rather informs it that it must verify with us first that his cache is still up-to-date
-								// This is used to handle ASF and user updates to WWW root, we don't want from the client to ever use outdated scripts
-								cacheControl.NoCache = true;
-
-								// All static files are public by definition, we don't have any authorization here
-								cacheControl.Public = true;
-
-								break;
-						}
+#if !NETFRAMEWORK && !NETSTANDARD
+		Dictionary<string, string> pluginsPaths = new();
+
+		if (PluginsCore.ActivePlugins?.Count > 0) {
+			foreach (IWebInterface plugin in PluginsCore.ActivePlugins.OfType<IWebInterface>()) {
+				if (string.IsNullOrEmpty(plugin.PhysicalPath) || string.IsNullOrEmpty(plugin.WebPath)) {
+					continue;
+				}
+
+				string staticFilesDirectory = Path.Combine(Path.GetDirectoryName(plugin.GetType().Assembly.Location)!, plugin.PhysicalPath);
 
-						ResponseHeaders headers = context.Context.Response.GetTypedHeaders();
+				if (Directory.Exists(staticFilesDirectory)) {
+					pluginsPaths.Add(staticFilesDirectory, plugin.WebPath);
```

Comment:
This will throw an `ArgumentException` in case multiple plugins declare the same directory. For example I can seea plugin developer having multiple plugins but developing only one GUI that supports all of them (which is being extracted by the plugins to a static location such as `/tmp/abrynos/ASF-www`)

Reply:
Since we're serving static files from the plugins location (`path/to/ASF/plugins/<pluginName>/path/to/www`), we don't have to worry about this.

Reply:
> Since we're serving static files from the plugins location [...]

This is how it is **_currently_** implemented. A plugin developer might disagree with this, which is why I brought up the idea of giving developers more freedom of choice.

Reply:
I agree, we should support relative and absolute paths if possible. The problem of not just using path as-is is because relative should be in regards to plugin's directory and not ASF config directory, which is how we usually treat relative paths with.

I think you can use `Path.IsPathRooted()` on physical path, if yes, just use it, if not, combine with assembly's location. This will initialize `staticFilesDirectory`, remaining logic with directory exists and everything is correct.